modules/notification/indexer/indexer.go

```diff
@@ -126,7 +125,11 @@ func (r *indexerNotifier) NotifyMigrateRepository(ctx context.Context, doer, u *
 }
 
 func (r *indexerNotifier) NotifyPushCommits(ctx context.Context, pusher *user_model.User, repo *repo_model.Repository, opts *repository.PushUpdateOptions, commits *repository.PushCommits) {
-	if setting.Indexer.RepoIndexerEnabled && opts.RefFullName == git.BranchPrefix+repo.DefaultBranch {
+	if !opts.RefFullName.IsBranch() {
+		return
+	}
+
+	if setting.Indexer.RepoIndexerEnabled && opts.RefFullName.BranchName() == repo.DefaultBranch {
```

Comment:
```suggestion
	if setting.Indexer.RepoIndexerEnabled && opts.RefFullName.IsBranchName(repo.DefaultBranch) {
```

Like this? Simpler than before and couldn't be misused.

Reply:
But sometimes, we need to send branch name as parameter but not compare the branch name. If we do that, we will have both `Branchname()` and `EqualBranchName()`.